File: code/java/ch14/practice/Warmup02RangeSumQuery.java
package ch14.practice;

import java.util.*;

/**
 * Warmup 2: Range Sum Query
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Given an array and queries [l, r], return the sum of
 *          arr[l..r] (inclusive, 0-indexed) for each query.
 *
 * EXAMPLES:
 *   solve([3,1,4,1,5,9], [[0,5],[2,4],[3,3]]) -> [23, 10, 1]
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup02RangeSumQuery {
    public static long[] solve(int[] arr, int[][] queries) {
        // TODO: Replace this with your solution
        return new long[queries.length];
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] arr = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int q = Integer.parseInt(sc.nextLine().trim());
        int[][] queries = new int[q][2];
        for (int i = 0; i < q; i++) {
            String[] parts = sc.nextLine().split(" ");
            queries[i][0] = Integer.parseInt(parts[0]);
            queries[i][1] = Integer.parseInt(parts[1]);
        }
        System.out.println(Arrays.toString(solve(arr, queries)));
        sc.close();
    }
}

File: code/cpp/ch14/practice/warmup_02_range_sum_query.cpp
/*
 * Warmup 2: Range Sum Query
 * Chapter 14: Prefix Sums — The Running Total Trick
 *
 * PROBLEM: Given array and queries [l, r], return sum of arr[l..r] for each.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <climits>
#include <iostream>
#include <vector>
using namespace std;

vector<long long> solve(vector<int> arr, vector<vector<int>> queries) {
    // TODO: Replace this with your solution
    return vector<long long>(queries.size(), 0);
}

// -- Do not change anything below this line --------------------------
int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];
    int q;
    cin >> q;
    vector<vector<int>> queries(q, vector<int>(2));
    for (int i = 0; i < q; i++) cin >> queries[i][0] >> queries[i][1];
    auto result = solve(arr, queries);
    for (int i = 0; i < (int)result.size(); i++)
        cout << result[i] << (i < (int)result.size()-1 ? " " : "\n");
    return 0;
}
